src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderBuyAsset.tsx

```diff
@@ -48,13 +52,49 @@ export const LimitOrderBuyAsset: React.FC<LimitOrderBuyAssetProps> = memo(
     selectedChainId,
     onSelectedChainIdChange,
   }) => {
+    const [isInputtingFiatSellAmount, setIsInputtingFiatSellAmount] = useState(false)
     const translate = useTranslate()
     const buyAssetSearch = useModal('buyTradeAssetSearch')
 
     const buyAmountCryptoPrecision = useAppSelector(selectBuyAmountCryptoPrecision)
     const buyAmountUserCurrency = useAppSelector(selectBuyAmountUserCurrency)
-
+    const sellAmountCryptoPrecision = useAppSelector(selectInputSellAmountCryptoPrecision)
     const manualReceiveAddress = useAppSelector(selectManualReceiveAddress)
+    const marketData = useAppSelector(state =>
+      selectMarketDataByAssetIdUserCurrency(state, asset.assetId),
+    )
+
+    const { setLimitPrice, setLimitPriceMode } = useActions(limitOrderInput.actions)
+
+    const handleToggleIsFiat = useCallback(() => {
+      setIsInputtingFiatSellAmount(prev => !prev)
+    }, [])
+
+    const handleAmountChange = useCallback(
+      (value: string) => {
+        // Always process the input change, even if value is 0
+        if (sellAmountCryptoPrecision && Number(sellAmountCryptoPrecision) > 0) {
```

Comment:
q: why Number cast

Reply:
Used bnOrZero in [85910a5](https://github.com/shapeshift/web/pull/8970/commits/85910a5ad18abe38e1bd411a103dba4dffef675f)